### ISSUE TITLE
Force student to type something before ending the exam

### ISSUE THREAD

**Issue by christianp** [numbas/Numbas#515]
We always have a handful of students who say that they ended an exam by mistake, because they didn't read the dialog messages.

We could force the students to engage their brains a bit by asking them to type something in a box before they're allowed to end. This could just be the word "end", or something unique to the exam so that they can't claim to have ended the wrong exam by mistake.

**Comment by NjeriNjoroge** [comment]
Hi, it's my first time contributing to open source and I would like to work on this issue.

**Comment by christianp** [comment]
Hi, thanks for offering!
First you should get the Numbas editor running on your computer, so you can test the code. Read our contribution guidelines at https://www.numbas.org.uk/contributing-to-numbas/#contributing-code.

I'll add a rough outline of what needs to be changed to the top of this issue.

**Comment by Redarcher9** [comment]
is this still open?
can i work on this?

**Comment by infoalisaa** [comment]
Hello!

 Is this issue still open?
 I am a first-time contributor, and I would like to help.

Thanks!

**Comment by christianp** [comment]
@infoalisaa Yes, it's still open. Open a pull request when you've got something for me to look at.

**Comment by abc22413** [comment]
Hi, is this issue still open?

**Comment by christianp** [comment]
@abc22413 yes, it is.

**Comment by vishwa58** [comment]
Hi I am looking at fixing this issue, but when I do a test run of the new question, I get a 404 error saying the index.html file does not exist. I followed the installation guide, and I was able to get it running, but I get the following error when I try to add a new question
[
<img width="1440" alt="Screen Shot 2021-04-19 at 5 53 21 PM" src="https://user-images.githubusercontent.com/50212303/115308487-c927c780-a138-11eb-9e99-3e6f8f690f4a.png">
](url)

**Comment by chanuka-dsc** [comment]
Hi, Is this issue still open ?. I am a first time contributer i would like to have go at this issue.
